nototools/gen_cplist.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/python
+#
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Generate lists of codepoints prefixed with 'u' or 'uni' from cmap data file.
+
+Occasionally the designers want data in this format to create lists of files
+for their tools, so let's just save this script.
+"""
+
+import argparse
+from os import path
+import sys
+
+from nototools import cmap_data
+from nototools import tool_utils
+
+
+def glyphstr(cp):
+  return ('uni%04x' % cp) if cp < 0x10000 else ('u%05x' % cp)
+
+
+def glyphstrs(cps):
+  return '\n'.join(glyphstr(cp) for cp in sorted(cps))
+
+
+def write_cp_list(cps, fname):
+  with open(fname, 'w') as f:
+    f.write(glyphstrs(cps))
+    f.write('\n')
+
+
+def generate_single(cmapdata, script, outfile):
+  for row in cmapdata.table.rows:
+    if script == row.script:
+      cps = tool_utils.parse_int_ranges(row.ranges)
+      write_cp_list(cps, outfile)
+      print >> sys.stderr, 'wrote %s to %s' % (script, outfile)
+      return
+  raise ValueError('no script "%s" in cmap data' % script)
+
+
+def generate(cmapdata, dst_dir, scripts, namepats):
+  if not scripts:
+    raise ValueError('no scripts')
+
+  if not namepats:
+    raise ValueError('no namepats')
+
+  if len(scripts) != len(namepats):
+    if len(namepats) != 1:
+      raise ValueError(
+          'Have %d script%s but %d namepats' %
+          (len(scripts), '' if len(scripts) == 1 else 's', len(namepats)))
+    if '%s' not in namepats[0] and len(scripts) > 1:
+      raise ValueError(
+          'Have multiple scripts but single namepat "%s" has no substitution'
+          % namepats[0])
+  if len(namepats) < len(scripts):
```

Comment:
We don't need this if statement, right? Just put the following line in the preceding block I think.